### PR TITLE
Bump CHaP and remove most constraints from the cabal file

### PR DESCRIPTION
**Pull request by michaelpj**
I revised a bunch of things in CHaP and the constraints are not needed
any more.

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -82,103 +82,18 @@ tests: True
 
 test-show-details: direct
 
-constraints:
-    hedgehog >= 1.0
-  , dependent-sum >= 0.7.1.0
-  , bimap >= 0.4.0
-  , libsystemd-journal >= 1.4.4
-  , systemd >= 2.3.0
```
Comment (michaelpj) [inline cabal.project:90]: moved to cabal file

**Comment by michaelpj** [comment]
Perhaps @newhoggy could review.

**Comment by michaelpj** [comment]
Maybe @disassembler ?

**Review by newhoggy — APPROVED** [review]

**Pull request merged**